.github/workflows/ruff.yml

```diff
@@ -0,0 +1,15 @@
+name: ruff
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+jobs:
+  ruff:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - run: pip install --user ruff
+    - run: ruff --format=github .
```

Comment:
You left in the flake8 workflow - do you think it is still needed?

Reply:
It is wired into https://github.com/reviewdog/action-flake8 and I cannot tell if you need `reviewdog`.

Reply:
Well, `reviewdoc` does the same as `ruff`, e.g. create annotation in the code, so if `ruff` is better suited, we don't need it anymore. I see that you have removed it already.
I may have a look at the differences between the two later, but your changes look good to me so far.
Will wait for @darcymason for his opinion on the change.

Reply:
Hi all, I have limited time right now, but I'll try to take a better look later today.

Reply:
@mrbean-bremen, I wasn't sure if there was anything else you wanted to review, but I'll merge in and we can adapt in other PR(s) as needed...

Reply:
That's ok, I'm happy with that. There is still #1776 for improvements.